> Would it be possible to have you add a line to that program attempting to add a double (like `+ 1.0`) to the _Float16? If so, we could remove the `PLATFORM(PLAYSTATION)` bit.

I believe the reason to add 1.0 is to guarantee the use of the variable so the compiler doesn't remove it in an optimization phase.

Simply adding `+ 1.0`, produced a warning on GCC12:

```
test-float16.cpp:3:16: warning: converting to ‘_Float16’ from ‘double’ with greater conversion rank
    3 |         foo += 1.0;
      |                ^~~
```

To avoid the warning message I added a `static_cast`.

--- a/Source/WTF/wtf/PlatformHave.h
+++ b/Source/WTF/wtf/PlatformHave.h
@@ -278,7 +278,7 @@
 #define HAVE_FLOAT16_INSTRUCTION 1
 #endif
 
-#if ((CPU(ARM64) || (CPU(X86_64) && !PLATFORM(PLAYSTATION))) && defined(__FLT16_MANT_DIG__))
+#if PLATFORM(COCOA) && (CPU(ARM64) || CPU(X86_64))
 #define HAVE_FLOAT16 1
 #endif
 

--- a/Source/cmake/WebKitCompilerFlags.cmake
+++ b/Source/cmake/WebKitCompilerFlags.cmake
@@ -460,6 +460,19 @@ int main() {
     unset(CMAKE_REQUIRED_FLAGS)
 endif ()
 
+if (NOT WTF_PLATFORM_COCOA)
+  set(FLOAT16_TEST_SOURCE "
+int main() {
+  _Float16 f;
+
+  f += static_cast<_Float16>(1.0);
+
+  return 0;
+}
+  ")
+  check_cxx_source_compiles("${FLOAT16_TEST_SOURCE}" HAVE_FLOAT16)
+endif ()
+
 if (CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND WTF_CPU_MIPS)
     # Work around https://gcc.gnu.org/bugzilla/show_bug.cgi?id=78176.
     # This only manifests when executing 32-bit code on a 64-bit

